### PR TITLE
Create HIP: Add no-op compatibility for cancun blobs

### PR DESCRIPTION
**Pull request by shemnon**
**Description:**:

HIP to add changes for EVM compatibility in the absence of transaction blobs.  Covers two new opcodes, `VERSIONEDHASH` and `BLOBBASEFEE`

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

This is one of four related Cancun HIPs.  Because of the broad coverage it was broken into 4 pieces.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: HIP/hip-866.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+interactions that are desirable to Hedera.
+
+Second, we need to support the new opcodes but ensure they operate as they would
+in the absence of any Blobs.
+
+## User stories
+
+* As a smart contract deployer, I want my smart contract that expects blob
+  support to not fail in unexpected ways.
+* As a smart contract user, I will not be able to successfully use contracts
```
Comment (Nana-EC) [inline HIP/hip-866.md:48]: not really a user story, can drop unless you can state it in an affirmative form of what a user should be able to do

File: HIP/hip-866.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+well-formed Type 3 transaction and provide different rejection messages. For now
+this will be a distinction without a difference as both valid and invalid Type 3
+transactions will be rejected.
+
+Second, we will begin to see calls to the simulation and estimation
+APIs (`eth_estimateGas`, `eth_call`) that may include fields indicating it is a
+blob transaction. In those cases we should similarly detect that a blob
+transaction is being simulated and reject the transaction before sending it to
+the simulation and estimation engines.
+
```
Comment (Nana-EC) [inline HIP/hip-866.md:101]: Q: Is it possibel for the mirror node [api/v1/contracts/call](https://testnet.mirrornode.hedera.com/api/v1/docs/#/contracts/contractsCall) to determine a type 3 transaction based on input params - I don't think so but I'm not sure.

If it is able to determine then you should add a `### Mirror Node web3 Module Implementation` that  will reject type 3 transactions

File: HIP/hip-866.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+## Backwards Compatibility
+
+The core EVM library shipping with Hedera as of version 0.46 already contains
+the needed EVM support. The activation will add a new Hedera EVM version that
+will activate all the Cancun support in one release. The generation of the call
+frame will need to be updated to set the BlobGasFee to 1. In prior EVM versions
```
Comment (Nana-EC) [inline HIP/hip-866.md:107]: Q: Does this mean there's a dependence on HIP 865?

Comment (shemnon) [reply HIP/hip-866.md:107]: The all depend on updated Besu EVM library code.  So not directly.  Although all four hips are all lightly cross-dependant.

File: HIP/hip-866.md (hunk 2 above)
Comment (shemnon) [reply HIP/hip-866.md:101]: Yes.  The first byte of the Ethereum Transaction is `0x03`.  Section added.

File: HIP/hip-866.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+in the absence of any Blobs.
+
+## User stories
+
+* As a smart contract deployer, I want my smart contract that expects blob
+  support to not fail in unexpected ways.
+* As a smart contract user, I will receive appropriate errors when I attempt to
+  introduce blobs via Ethereum transactions.
+* As a future Hedera Core developer, I want the option to provide blob like
+  behavior using EIP-4844 semantics and existing opcodes.
```
Comment (david-bakin-sl) [inline HIP/hip-866.md:51]: "existing opcodes": this must mean "new opcodes defined in EIP-484"?  (In the context here it at first seemed to be referring to opcodes that exist _prior_ to Cancun.

**Review by david-bakin-sl — DISMISSED** [review]
LGTM

File: HIP/hip-866.md (hunk 4 above)
Comment (shemnon) [reply HIP/hip-866.md]: changed to 'the related opcodes'

File: HIP/hip-866.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+In order to both preserve compatibility and future design space we need to act
+as if blobs are not being added to Hedera. This allows existing contracts that
+may depend on blob behavior to continue to function in the absence of blobs.
+
+First we will prevent blobs from entering the system. There are complex EL/CL
+interactions in Ethereum Mainnet, but as far as Hedera's consensus is similar to
+Ethereum Mainnet's, the main entry point is a "Type 3" transaction that allows
+blobs to be attached. Furthermore, Type 3 transactions require a blob to be
+attached. Prohibiting Type 3 transactions will thus be sufficient to keep blobs
+form entering the EVM's realm of concern and similarly will not prohibit other
```
Comment (Nana-EC) [inline HIP/hip-866.md:38]:
```suggestion
from entering the EVM's realm of concern and similarly will not prohibit other
```

**Review by Nana-EC — APPROVED** [review]
LG.
1 spelling issue

**Review by Nana-EC — APPROVED** [review]
LG

**Review by david-bakin-sl — APPROVED** [review]
LGTM recheck

**Pull request merged**